subprojects/docs/src/docs/userguide/authoring-builds/introduction/writing_build_scripts.adoc

```diff
@@ -0,0 +1,538 @@
+}
+
+application {
+    mainClass = 'main'                                                  // <6>
+}
+
+tasks.named('test') {
+    useJUnitPlatform()                                                  // <7>
+}
+----
+<1> Apply the org.jetbrains.kotlin.jvm Plugin to add support for Kotlin.
+<2> Apply the application plugin to add support for building a CLI application in Java.
+<3> Use Maven Central for resolving dependencies.
+<4> Use the link:https://mvnrepository.com/artifact/org.jetbrains.kotlin/kotlin-test-junit5[Kotlin JUnit 5] integration, link:https://mvnrepository.com/artifact/org.junit.jupiter/junit-jupiter-engine[JUnit engine], and link:https://mvnrepository.com/artifact/org.junit.platform/junit-platform-launcher[JUnit 5] integration for testing code.
+<5> Use the link:https://mvnrepository.com/artifact/com.google.guava/guava[Guava] dependency from Google for the application.
+<6> Define the main class for the application.
+<7> Use JUnit Platform, a JVM testing framework, for unit tests.
+=====
+====
+
+=== Add extensions via plugins
+
+Plugins are used to extend Gradle. They are also used to modularize and reuse project configuration.
+
+Plugins can be applied using the `PluginAware.apply(java.util.Map)` method, or by using the `PluginDependenciesSpec` plugins script block:
+
+[source]
+----
+plugins {
+    id("org.jetbrains.kotlin.jvm") version "1.9.0"
+    application
+}
+----
+
+In the example, the `application` plugin, which is included with Gradle, has been applied, describing our project as a Java application.
+
+The Kotlin gradle plugin, version 1.9.0 has also been applied.
+This plugin is not included with Gradle and therefore has to be described using a `plugin id` and a `plugin version` so that Gradle can find and apply it.
+
+=== Add dependencies and dependency location
+
+A project generally has a number of dependencies it needs in order to do its work.
+
+Those dependencies are grouped in configurations, and can be retrieved and uploaded from repositories.
+
+Use the `DependencyHandler` returned by `Project.getDependencies()` method to manage the dependencies.
+Use the `RepositoryHandler` returned by `Project.getRepositories()` method to manage the repositories.
+
+[source]
+----
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("org.jetbrains.kotlin:kotlin-test-junit5")
+    testImplementation("org.junit.jupiter:junit-jupiter-engine:5.9.3")
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+    implementation("com.google.guava:guava:32.1.1-jre")
+}
+----
+
+=== Use convention properties
+
+A plugin can add properties and method to a project through the project's `Convention` object.
+
+In the example, the `application` plugin added an `application` property which is used to detail the main class of our Java application:
+
+[source]
+----
+application {
+    mainClass.set("main")
+}
+----
+
+=== Create or update tasks
```

Comment:
Note: Wondering on the use of "create" in the title

Also - would be worth mentioning somewhere that the "create" method exists, but is no longer recommended and link to the section talking about it more.